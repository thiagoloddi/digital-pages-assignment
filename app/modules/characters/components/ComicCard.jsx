import React from 'react';
import injectSheet from 'react-jss';
import Paper from '@material-ui/core/Paper';
import Image from '../../../components/ui/Image';

const style = {
  paper: {
    padding: '15px',
    marginTop: '20px',
    '& > *': {
      display: 'inline-block'
    }
  },
  image: {
    height: '200px',
    width: '200px'
  },
  info: {
    verticalAlign: 'top',
    paddingLeft: '20px',
    width: 'calc(100% - 200px)',
    fontSize: '18px',
    '& > *': {
      display: 'inline-block'
    }
  },
  right: {
    float: 'right'
  }
};

const ComicCard = ({ title, issueNumber, thumbnail, description, classes }) => {
  
  return (
    <Paper className={classes.paper}>
      <Image src={`${thumbnail.path}.${thumbnail.extension}`} className={classes.image}/>
      <div className={classes.info}>
        <span><strong>TÍTULO: </strong>{title}</span>
        <span className={classes.right}><strong>NÚMERO DE CAPA: </strong>{issueNumber}</span>
        <p>{description}</p>
      </div>
    </Paper>
  )
}

export default injectSheet(style)(ComicCard);
