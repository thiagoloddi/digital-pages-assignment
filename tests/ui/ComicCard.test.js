import React from 'react';
import renderer from 'react-test-renderer';
import ComicCard from '../../app/modules/characters/components/ComicCard';

const createNodeMock = e => {
  if(e.type == 'img') {
    return {};
  }
};

it('renders correctly', () => {
  const tree = renderer
    .create(<ComicCard title="A Title" issueNumber="1" thumbnail={{ path: "path", extensions: ";jpg" }} description="A description" />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});