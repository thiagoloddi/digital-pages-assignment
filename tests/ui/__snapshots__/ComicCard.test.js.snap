// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 ComicCard-paper-0-1-1"
>
  <div
    className="Image-container-0-1-5 ComicCard-image-0-1-2"
  >
    <img
      className="Image-img-0-1-6"
      src="path.undefined"
      style={
        Object {
          "visibility": "hidden ",
        }
      }
    />
    <img
      className="Image-loading-0-1-7"
      src="/assets/progress.gif"
      style={null}
    />
  </div>
  <div
    className="ComicCard-info-0-1-3"
  >
    <span>
      <strong>
        TÍTULO: 
      </strong>
      A Title
    </span>
    <span
      className="ComicCard-right-0-1-4"
    >
      <strong>
        NÚMERO DE CAPA: 
      </strong>
      1
    </span>
    <p>
      A description
    </p>
  </div>
</div>
`;
